File: frontend/web/src/api/listings/getListings.ts
import axiosInstance from "../../utils/AxiosInstance.ts"

const getListings = async () => {
    const token = localStorage.getItem("token");

    const { data } = await axiosInstance.get("/api/listings", {
        headers: {
            Authorization: `Bearer ${token}`,
        }
    });

    return data.listings;
}

export default getListings;

File: frontend/desktop/python-client/PyQt5App/api/listings/getListings.py
import requests
from PyQt5App.utils.Address import address
from PyQt5App.utils.sessionManager import SessionManager

class GetListings:
    def __init__(self):
        self.address = address()
        self.sessionManager = SessionManager()
        self.token = self.sessionManager.get_token()

    def getListings(self):
        response = requests.get(
            f"{self.address}/api/listings",
            headers={
                "Authorization": f"{self.token}",
            }
        )
        data = response.json()
        return data.get("listings", [])